plugins/inputs/syslog/syslog.go

```diff
@@ -0,0 +1,378 @@
+package syslog
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/go-syslog/rfc5424"
+	"github.com/influxdata/go-syslog/rfc5425"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	tlsConfig "github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const defaultReadTimeout = time.Millisecond * 500
+const ipMaxPacketSize = 64 * 1024
+
+// Syslog is a syslog plugin
+type Syslog struct {
+	Address  string `toml:"server"`
+	Protocol string
+	tlsConfig.ServerConfig
+	KeepAlivePeriod *internal.Duration
+	ReadTimeout     *internal.Duration
+	MaxConnections  int
+	BestEffort      bool
+
+	now func() time.Time
+
+	mu sync.Mutex
+	wg sync.WaitGroup
+	io.Closer
+
+	isTCP         bool
+	tcpListener   net.Listener
+	tlsConfig     *tls.Config
+	connections   map[string]net.Conn
+	connectionsMu sync.Mutex
+
+	udpListener net.PacketConn
+}
+
+var sampleConfig = `
+    ## Specify an ip or hostname with port - eg., localhost:6514, 10.0.0.1:6514
+    ## Address and port to host the syslog receiver.
+    ## If no server is specified, then localhost is used as the host.
+    ## If no port is specified, 6514 is used (RFC5425#section-4.1).
+	server = ":6514"
+	
+	## Protocol (default = tcp)
+	## Should be one of the following values:
+	## tcp, tcp4, tcp6, unix, unixpacket, udp, udp4, udp6, ip, ip4, ip6, unixgram.
+	## Otherwise forced to the default.
+	# protocol = "tcp"
+
+    ## TLS Config
```

Comment:
This needs to be indented to the same level as the rest of the config

Reply:
Done.